samza-core/src/main/scala/org/apache/samza/job/JobRunner.scala

```diff
@@ -118,6 +118,21 @@ class JobRunner(config: Config) extends Logging {
     }
     coordinatorSystemProducer.stop()
 
+
+    // if diagnostics is enabled, create diagnostics stream as specified in config
+    if (new JobConfig(config).getDiagnosticsEnabled) {
+      val systemStreamName = new MetricsConfig(config).
+        getMetricsReporterStream(MetricsConfig.METRICS_SNAPSHOT_REPORTER_NAME_FOR_DIAGNOSTICS).get
+      val systemStream = StreamUtil.getSystemStreamFromNames(systemStreamName)
+      val diagnosticStreamSpec = new StreamSpec("diagnostics-stream-id", systemStream.getStream, systemStream.getSystem)
+      val diagnosticSysAdmin = systemFactory.getAdmin(systemStream.getSystem, config)
```

Comment:
This "systemFactory" comes from the coordinator system. However, it does not look like it is guaranteed that systemStream.getSystem is the same as the coordinator system, so I don't think you can always use "systemFactory" here.